### DISCUSSION TITLE
ServiceClient RetrieveMultiple returns records over 5000?

### DISCUSSION THREAD

**Discussion by rgx91** [microsoft/PowerPlatform-DataverseServiceClient#334] [Q&A] [answered]
Hello,
as the title suggests, I wondering the Dataverse client ServiceClient returns rows over 5000 limit, or we need to use fetchXml and pagingCookie if the result set is over 5000 records?

**Comment by BetimBeja** [comment, marked as answer]
AFAIK the 5000 records limit is a service limit, not a client limit. 
Also, pagination can be done with `QueryExpression` also, why do you say `we need to use fetchXml...` 
https://learn.microsoft.com/en-us/power-apps/developer/data-platform/org-service/page-large-result-sets-with-queryexpression?WT.mc_id=DX-MVP-5004767
